### PR TITLE
Add integration test field on Feature Review page

### PR DESCRIPTION
**Pull request by popovm**
This pull request adds a field to the Feature Review page, which displays the status of the integration test build.
screen shot 2017-09-08 at 18 11 57

### CODE REVIEW

**Comment by popovm** [comment]
@FundingCircle/prod-ops @bliof @0mega  👀

File: app/models/events/integration_test_event.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  class IntegrationTestEvent < Events::BaseEvent
+    def app_name
+      details.fetch('app_name')
+    end
+
+    def version
+      details.fetch('version', 'unknown')
+    end
+
+    def url
```
Comment (bliof) [inline app/models/events/integration_test_event.rb]: maybe `details_url` will be better

**Comment by 0mega** [comment]
@sgerrand there might be some confusion with what the name represents. My understanding was that the `IntegrationTest` will represent the tests that are run on Jenkins for our apps. So it's part of the core model of shipment tracker to represent other application's result of running the integration tests, as opposed to tests that we run against shipment tracker code.

**Comment by sgerrand** [comment]
Whoops, apologies. 😊

File: app/models/integration_test.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+# frozen_string_literal: true
+class IntegrationTest
```
Comment (sgerrand) [inline app/models/integration_test.rb]: Suggest `IntegrationTestResult` as a more explicit name.

**Comment by popovm** [comment]
We discussed with @bliof and we agreed that actually there is no reason of a new event type.
We can re-use the `BuildEvent`, as this is actually a build and just have a new type of `source`, which states that this is a integration tests build.

**Comment by popovm** [comment]
@bliof @0mega That's ready for review again 👀

File: app/models/repositories/build_repository.rb [hunk 3]
```diff
@@ -6,22 +6,38 @@
...
 
-    def builds_for(apps:, at: nil)
+    def integration_tests_result_for(versions:, at: nil)
+      created_at_query = at ? store.arel_table['event_created_at'].lteq(at) : nil
+
+      store
+        .where(version: versions, build_type: 'integration')
+        .where(created_at_query)
+        .order('id DESC')
+        .first
```
Comment (bliof) [inline app/models/repositories/build_repository.rb]: Let's allow multiple builds like the one for unit tests.

**Comment by popovm** [comment]
@mbdimitrova @bliof Could you take a final look at this pull request 👀

File: spec/decorators/feature_review_with_statuses_spec.rb [hunk 4]
```diff
@@ -118,51 +123,84 @@
...
     end
+  end
 
-    context 'when there are no builds' do
-      let(:builds) { {} }
+  describe '#integration_test_result_status' do
+    context 'when all integration test results pass' do
+      let(:integration_test_results) do
+        {
+          'frontend' => Build.new(success: true),
```
Comment (bliof) [inline spec/decorators/feature_review_with_statuses_spec.rb]: It would be better to add a factory for `unit_test_build` and `integration_test_build`.

Comment (popovm) [reply spec/decorators/feature_review_with_statuses_spec.rb]: @bliof Unfortunately, `Build` is not a `ActiveRecord`, but a `Virtus` object. That's why I am not sure we can use factory_girl for building `builds`

**Comment by popovm** [comment]
@bliof I added a `BuildHelper` for constructing builds.

File: spec/decorators/feature_review_with_statuses_spec.rb (hunk 4 above)
Comment (sgerrand) [reply spec/decorators/feature_review_with_statuses_spec.rb]: Factory Girl can build any kind of Ruby object, including POROs. 😉

File: spec/support/build_helper.rb [hunk 5]
```diff
@@ -0,0 +1,13 @@
+# frozen_string_literal: true
+
+module Support
+  module BuildHelper
+    def build_unit_test_build(options = {})
```
Comment (sgerrand) [inline spec/support/build_helper.rb]: 🤔  Please use a factory for this instead. It's simpler to understand how and where test objects are constructed if it all happens in the one place.

e.g.
```ruby
FactoryGirl.define do
  factory :build do
    factory :integration_test_build do
      build_type: 'integration'
    end

    factory :unit_test_build do
      build_type: 'unit'
    end
  end
end
```

**Comment by popovm** [comment]
@sgerrand I added factories for builds and removed the previous helper. 🙂

**Pull request merged**